server.go

```diff
@@ -90,32 +89,32 @@ func (s *Server) Shutdown(ctx context.Context) error {
 }
 
 func (s *Server) handleMh(w http.ResponseWriter, r *http.Request) {
+	start := time.Now()
+	ws := newResponseWriterWithStatus(w)
```

Comment:
Please avoid defining `ws` inside the method, pass it as an argument instead. One might accidentally use `w` instead of `ws` that would result into status code being incorrectly captured.

Reply:
Ok, misread the code, all good